### DISCUSSION TITLE
Starting the VoiceCraft server on Linux.

### DISCUSSION THREAD

**Discussion by SineVector241** [AvionBlock/VoiceCraft#101] [General]
1. Download the VoiceCraft.Server.zip from the **Releases** page.
2. Unzip the file.
3. Install dotnet on linux by running `sudo apt-get install dotnet-runtime-8.0` (ubuntu)
4. Navigate the terminal to inside the unzipped VoiceCraft.Server folder and run `dotnet VoiceCraft.Server.dll`

Should run if nothing goes wrong.

**Comment by Miniontoby** [comment]
New alternative instructions that should be added to the main message:


# Running with docker

If you want to use docker instead, then you probably already have docker installed.

Images are provided on docker.io and on the github packages, so these would be the 'url' for `docker pull`:
- [`sinevector241/voicecraft`](https://hub.docker.com/r/sinevector241/voicecraft/tags)
- [`ghcr.io/sinevector241/voicecraft`](https://github.com/SineVector241/VoiceCraft-MCBE_Proximity_Chat_Docker/pkgs/container/voicecraft)

They both started at `1.0.6`, older versions are NOT supported (yet).
And you can always use the `latest` tag to get the newest version of voicecraft.

## Example

You can start a server using this command:
```bash
docker run -d -p 9050:9050 --name voicecraft_server sinevector241/voicecraft:latest
# OR
docker run -d -p 9050:9050 --name voicecraft_server ghcr.io/sinevector241/voicecraft:latest
```

## Editing the config

If you need to edit the config, then you can copy the config file to your local machine, edit it, and then upload it again:
```bash
# Download the file to local folder
docker cp voicecraft_server:config/ServerProperties.json .

# Edit the file using your editor (mine is nano)
nano ServerProperties.json

# Once you're done, upload it again and restart the server
docker cp ServerProperties.json voicecraft_server:config/ServerProperties.json
docker restart voicecraft_server
```
